infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -139,22 +140,30 @@ public KubernetesInternalRuntime(
     this.executor = sharedPool.getExecutor();
     this.runtimeStates = runtimeStates;
     this.machines = machines;
+    this.startSynchronizer = new StartSynchronizer(context.getIdentity(), eventService);
```

Comment:
Side note: looks like event service is just a dependency of StartSynchronizer and KubernetesInternalRuntime doesn't have to have this dependency too.  So it might be cleaner to create some factory for this class and hide its dependencies.

Reply:
I thought about it, and while there is only one injected field creating one more class looks a bit overkill. I would prefer to leave it as it is for now.